source/s3_meta_request.c

```diff
@@ -1711,6 +1711,22 @@ static void s_s3_meta_request_event_delivery_task(struct aws_task *task, void *a
                 aws_atomic_fetch_sub(&client->stats.num_requests_streaming_response, 1);
 
                 ++num_parts_delivered;
+
+                if (request->send_data.metrics != NULL) {
+                    /* Request is done streaming the body, complete the metrics for the request now. */
+                    struct aws_s3_request_metrics *metrics = request->send_data.metrics;
+                    metrics->crt_info_metrics.error_code = error_code;
```

Comment:
why are you altering the metric's error_code? the `error_code` on the right-hand-side is whether the meta-request has failed. I would think the metrics callback would want to report whether that individual HTTP request failed.